### ISSUE TITLE
Mint concept:ecma262-number-format to retire rfc8785-number-serialization-non-ecma262 loss

### ISSUE THREAD

**Issue by TSavo** [TSavo/provekit#1367]
## Summary

Both JCS shims (rust serde_json + java jackson) currently declare \`loss = [\"rfc8785-number-serialization-non-ecma262\"]\` on \`concept:rfc8785-jcs-encode\` + \`concept:rfc8785-jcs-encode-value\`.

The loss is real: \`serde_json::Number::to_string()\` (rust) and \`JsonNode.asText()\` (java) emit a chosen textual form that's NOT ECMA-262 §7.1.12.1 (which RFC 8785 §3.2.2.3 requires).

## Resolution

Mint \`concept:ecma262-number-format\` as a substrate-canonical primitive operation:

\`\`\`
concept:ecma262-number-format(value: Number) -> String
  = ECMA-262 7.1.12.1 ToString-applied-to-Number
\`\`\`

Then route number serialization in both JCS encoders through the new concept. The \`rfc8785-number-serialization-non-ecma262\` loss declaration can then be removed from the shim sources + body-templates.

## Background

Surfaced during PR #1365 reviewer feedback (CodeRabbit Major + Codex P1). The loss was originally undeclared which contradicted the substrate-honesty thesis (\"supra omnia, rectum\"); PR #1365 declared the loss as a stopgap. This issue tracks closing the actual gap.

## Test plan

- [ ] Mint \`concept:ecma262-number-format\` with proper formal_sort.
- [ ] Add per-language realizations (rust: dtoa or proper format; java: existing ECMA-262 number lib or hand-rolled).
- [ ] Update both JCS shims to use the new concept.
- [ ] Remove \`rfc8785-number-serialization-non-ecma262\` from \`loss\` declarations.
- [ ] Verify JCS output against an RFC 8785 test vector with edge-case numbers (1e21, 0.1, NaN handling, etc.).

**Comment by TSavo** [comment]
Issue bankruptcy 2026-07-01: closing all open issues to start fresh. The current state of record is the design-crimes remediation plan (docs/superpowers/plans/2026-07-01-design-crimes-remediation.md, PR #2979) and the audits in flight. If this issue is still real, it will resurface through an instrument.
